apsuite/orbcorr/si_calc_bump_matrix_at_c2_magnet.py

```diff
@@ -0,0 +1,136 @@
+"""."""
+
+import numpy as np
+import matplotlib.pyplot as mplt
+import matplotlib.gridspec as mgs
+
+from pymodels import si
+import pyaccel
+from . import OrbitCorr
+
+
+def calc_matrices(minsingval=0.2, mb2idx=0, deltax=10e-6):
+    """."""
+    # NOTE: valid only for first B2 in sector!
+    mod = si.create_accelerator()
+    orbcorr = OrbitCorr(mod, 'SI')
+    orbcorr.params.tolerance = 1e-9
+    orbcorr.params.minsingval = minsingval
+
+    mb2idx = max(min(mb2idx, 19), 0)
+    mb2i = pyaccel.lattice.find_indices(
+        orbcorr.respm.model, 'fam_name', 'mb2')[2*mb2idx]
+
+    orb0 = orbcorr.get_orbit()
+    kicks0 = orbcorr.get_kicks()
+
+    # idcs = np.array([1, 2, 160+1, 160+2])
+    idcs = np.array([2, 3, 160+2, 160+3])
+    idcs += 8*mb2idx
+
+    # remove corrs between BPMs
+    orbcorr.params.enbllistch[mb2idx*6 + 2] = False
+    orbcorr.params.enbllistcv[mb2idx*8 + 2] = False
+    orbcorr.params.enbllistcv[mb2idx*8 + 3] = False
+
+    matb2 = np.zeros((4, 4), dtype=float)
+    mator = np.zeros((4, 4), dtype=float)
+    for i, idx in enumerate(idcs):
+        gorb = orb0.copy()
+        orbcorr.set_kicks(kicks0)
+        gorb[idx] += deltax/2
+        orbcorr.correct_orbit(goal_orbit=gorb)
+        orbp = orbcorr.get_orbit()[idcs]
+        b2p = pyaccel.tracking.find_orbit6(
+            orbcorr.respm.model, indices='open')[0:4, mb2i - 10]
```

Comment:
maybe define a variable for this `10` here?

Reply:
done